ait/core/server/server.py

```diff
@@ -0,0 +1,255 @@
+        err_msgs = {}
+
+        for stream_type in ['inbound', 'outbound']:
+            err_msgs[stream_type] = common_err_msg.format(stream_type) + specific_err_msg[stream_type]
+            streams = ait.config.get('server.{}-streams'.format(stream_type))
+
+            if streams is None:
+                log.warn(err_msgs[stream_type])
+            else:
+                for index, s in enumerate(streams):
+                    try:
+                        strm = self._create_stream(s['stream'], stream_type)
+                        if stream_type == 'inbound' and type(strm) == PortInputStream:
+                            self.servers.append(strm)
+                        elif stream_type == 'inbound':
+                            self.inbound_streams.append(strm)
+                        elif stream_type == 'outbound':
+                            self.outbound_streams.append(strm)
+                        log.info('Added {} stream {}'.format(stream_type, strm))
+                    except Exception:
+                        exc_type, value, tb = sys.exc_info()
+                        log.error('{} creating {} stream {}: {}'.format(exc_type,
+                                                                        stream_type,
+                                                                        index,
+                                                                        value))
+        if not self.inbound_streams and not self.servers:
+            log.warn(err_msgs['inbound'])
+
+        if not self.outbound_streams:
+            log.warn(err_msgs['outbound'])
+
+    def _create_stream(self, config, stream_type):
+        """
+        Creates a stream from its config.
+
+        Params:
+            config:       stream configuration as read by ait.config
+            stream_type:  either 'inbound' or 'outbound'
+        Returns:
+            stream:       a Stream
+        Raises:
+            ValueError:   if any of the required config values are missing
+        """
+        if stream_type not in ['inbound', 'outbound']:
+            raise ValueError('Stream type must be \'inbound\' or \'outbound\'.')
+
+        if config is None:
+            raise ValueError('No stream config to create stream from.')
+
+        name = config.get('name', None)
+        if name is None:
+            raise(cfg.AitConfigMissing(stream_type + ' stream name'))
+        if name in [x.name for x in (self.outbound_streams +
+                                     self.inbound_streams +
+                                     self.servers +
+                                     self.plugins)]:
+            raise ValueError('Stream name already exists. Please rename.')
+
+        stream_input = config.get('input', None)
+        if stream_input is None and stream_type is 'inbound':
+            raise(cfg.AitConfigMissing(stream_type + ' stream input'))
+
+        stream_handlers = [ ]
+        if 'handlers' in config:
```

Comment:
We've got a lot going on in this function and it looks like there's good places (here / below, but also others) to split code out into functions that helps us keep stuff readable / testable.